modules/scanning-delegate-role/main.tf

```diff
@@ -539,7 +542,146 @@ resource "aws_iam_role_policy_attachment" "worker_attachment" {
 }
 
 resource "aws_iam_role_policy_attachment" "workers_dspm_attachment" {
-  count      = var.sensitive_data_scanning_enabled ? 1 : 0
+  count      = length(aws_iam_policy.scanning_worker_dspm_policy)
   policy_arn = aws_iam_policy.scanning_worker_dspm_policy[0].arn
   role       = aws_iam_role.role.name
 }
+
+// RDS Specific resources
+
+// RDS Service Role for S3 Exports
+data "aws_iam_policy_document" "rds_service_role_assume_policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    effect  = "Allow"
+    principals {
+      type        = "Service"
+      identifiers = ["export.rds.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "rds_service_role_policy_document" {
+  statement {
+    sid = "DatadogAgentlessScannerExportPolicy"
+    actions = [
+      "s3:PutObject*",
+      "s3:ListBucket",
+      "s3:GetObject*",
+      "s3:DeleteObject*",
+      "s3:GetBucketLocation"
+    ]
+    resources = [
+      "arn:aws:s3:::datadog-agentless-scanning-*",
```

Comment:
I assume you're doing this because you don't know the name of the bucket in advance. Is there a way to retrieve it and use it in the policy?

Reply:
The bucket depends on the delegate role. I don't want to bring that extra level of complexity (at module and deployment level) where both modules depend on each other. It gets worse. We deploy the delegate role once per account and the buckets are deployed once per region. We might need to add more regions/buckets after the deployment and that would mean updating the delegate role as well. We might revisit this later to check whether it's doable but for now, we prefer to keep it this way.